connectivity/manifests/allow-all-except-world.yaml

```diff
@@ -2,12 +2,20 @@ apiVersion: cilium.io/v2
 kind: CiliumNetworkPolicy
 metadata:
   namespace: cilium-test
-  name: allow-all
+  name: allow-all-except-world
 spec:
   endpointSelector: {}
   egress:
   - toEntities:
-    - all
+    - host
+    - remote-node
+    - cluster
+    - init
+    - health
```

Comment:
This looks like a different test to me. The point of `allow-all` test was to test with a policy that does not drop anything. This makes it possible to rule out policy implementation as the cause for a consistent failure one might see in a deployment.

Also, I'm not sure these listed entities will cover any CIDRs, such as generated from toFQDN policies.

Reply:
> The point of allow-all test was to test with a policy that does not drop anything.

Doesn't having no policies loaded at all implement the same? We will also go through the policy engine and allow all.

> This makes it possible to rule out policy implementation as the cause for a consistent failure one might see in a deployment.

Why doesn't `no-policies` already allow us to rule out policy enforcement as the cause of failures?

Reply:
>Doesn't having no policies loaded at all implement the same? We will also go through the policy engine and allow all.

If an endpoint isn't selected by any policies, we won't enter the policy enforcement block at all, no?

Reply:
>If traffic cannot be identified, then the datapath falls back to assigning it as "world".

This behavior will try to mask failures where packets are not taking the expected path, no? Why don't we distinguish between "world" and "unknown"?

>Also, I'm not sure these listed entities will cover any CIDRs, such as generated from toFQDN policies.

+1

Reply:
> If an endpoint isn't selected by any policies, we won't enter the policy enforcement block at all, no?

We will. That's why we load two rules in that case to allow everything on egress and ingress (+ allow all from hostns on ingress):
```
$ vagrant ssh k8s1 -- sudo cilium bpf policy get --all
/sys/fs/bpf/tc/globals/cilium_policy_00980:

POLICY   DIRECTION   LABELS (source:key[=value])   PORT/PROTO   PROXY PORT   BYTES   PACKETS   
Allow    Ingress     reserved:unknown              ANY          NONE         0       0         
Allow    Ingress     reserved:host                 ANY          NONE         0       0         
Allow    Egress      reserved:unknown              ANY          NONE         0       0
[...]
```

Reply:
> Why doesn't no-policies already allow us to rule out policy enforcement as the cause of failures?

It does, and the actual policy in the datapath should be the same :-) So yes it seems we do not need the `allow-all` test, so it can be morphed into a more restrictive test.